pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,428 @@
+        ("res3",                 c_uint8, 1),
+        ("SRQ_PON",              c_uint8, 1),
+
+        #Byte 4: Error Information
+        ("ERR_cal",        c_uint8, 1),
+        ("ERR_RAM",        c_uint8, 1),
+        ("ERR_ROM",        c_uint8, 1),
+        ("ERR_slope",      c_uint8, 1),
+        ("ERR_AD",         c_uint8, 1),
+        ("ERR_AD_Link",    c_uint8, 1),
+        ("res4",           c_uint8, 1),
+        ("res5",           c_uint8, 1),
+
+        #Byte 5: DAC Value
+        ("DAC_value",       c_uint8, 8),
+    ]
+    def __str__(self):
+        """
+        Returns a pretty formatted (human readable) string showing the status of the instrument
+
+        """
+        inv_modes = dict(zip(MODES.values(),MODES.keys()))
+        cur_mode = inv_modes["F" + str(self.function)]
+        cur_range = list(RANGES[cur_mode].keys())[self.range - 1]
+        if cur_range >= 1E6:
+            r_str = str(cur_range / 1E6) + ' M'
+        elif cur_range >= 1000:
+            r_str = str(cur_range / 1000) + ' k'
+        elif cur_range <= 1:
+            r_str=str(cur_range * 1000) + ' m'
+        else:
+            r_str=str(cur_range) + ' '
+        return (
+            "function: {}, range: {}, digits: {}\
+                \nStatus:\n  internal | external trigger: {} | {}\n  Auto ranging: {}\n  AutoZero: {}\
+                \n  50Hz mode: {}\n  Front/Rear selection: {}\n  Calibration enable: {}\
+                \nSerial poll mask (SRQ):\n  SRQ for Data ready: {}\
+                \n  SRQ for Syntax error: {}\n  SRQ for Internal error: {}\n  SRQ Front Panel button: {}\
+                \n  SRQ for Cal err: {}\n  SQR for Power on: {}\
+                \nError information: \n Calibration: {}  RAM: {}  ROM: {}  AD Slope: {} AD: {} AD-Link: {} \
+                \nDAC value: {}".format(
+                cur_mode, r_str, 6-self.digits, self.int_trig, self.ext_trig, self.auto_range,
+                self.auto_zero, self.fifty_hz, self.front_rear , self.cal_enable,
+                self.SRQ_data_rdy, self.SRQ_syntax_error, self.SRQ_internal_error, self.SRQ_front_panel, self.SRQ_cal_error, self.SRQ_PON,
+                self.ERR_cal, self.ERR_RAM, self.ERR_ROM, self.ERR_slope, self.ERR_AD, self.ERR_AD_Link,
+                self.DAC_value)
+            )
+
+class Status(ctypes.Union):
+    """Union type element for the decoding of the status bit-fields"""
+    _fields_ = [
+        ("B", Status_bytes),
+        ("b", Status_bits)
+    ]
+
+# @staticmethod does not work ?!
+def get_status(status_bytes, field=None):
+    """Method to handle the decoding of the status bytes into something meaningfull..
+
+    :param status_bytes:    list of bytes to be decoded
+    :return ret_val:
+
+    """
+    ret_val = Status(Status_bytes(*status_bytes))
+    if field is None:
+        return ret_val.b
+    else:
+        if field == "SRQ":
+            return getattr(ret_val.B, "byte3")
+        else:
+            return getattr(ret_val.b, field)
```

Comment:
Can you not equally well have a single if - elif - else statement rather than a nested if-else statement?

Reply:
Thank you, looking at the code again after a while, this make totally sense and looks better.